From: David Hildenbrand <david@redhat.com>
Subject: selftest-mm-ksm_functional_tests-add-prot_none-test-fix
Date: Mon, 7 Aug 2023 17:33:54 +0200

As noted by Peter, we should be using sizeof(i) in test_prot_none().

Link: https://lkml.kernel.org/r/e9cdb144-70c7-6596-2377-e675635c94e0@redhat.com
Signed-off-by: David Hildenbrand <david@redhat.com>
Cc: Peter Xu <peterx@redhat.com>
Signed-off-by: Andrew Morton <akpm@linux-foundation.org>
---

 tools/testing/selftests/mm/ksm_functional_tests.c |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

--- a/tools/testing/selftests/mm/ksm_functional_tests.c~selftest-mm-ksm_functional_tests-add-prot_none-test-fix
+++ a/tools/testing/selftests/mm/ksm_functional_tests.c
@@ -516,7 +516,7 @@ static void test_prot_none(void)
 	/* Store a unique value in each page on one half using ptrace */
 	for (i = 0; i < size / 2; i += pagesize) {
 		lseek(mem_fd, (uintptr_t) map + i, SEEK_SET);
-		if (write(mem_fd, &i, sizeof(size)) != sizeof(size)) {
+		if (write(mem_fd, &i, sizeof(i)) != sizeof(i)) {
 			ksft_test_result_fail("ptrace write failed\n");
 			goto unmap;
 		}
_
